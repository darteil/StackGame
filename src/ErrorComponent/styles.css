.wrap {
    width: 100%;
    min-height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background-color: #0E0F1A;
}

.text p {
    color: #a0a4a7;
    font-size: 30px;
}
