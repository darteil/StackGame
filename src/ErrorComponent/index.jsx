import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.css';

const ErrorComponent = props => (
  <div className={styles.wrap}>
    <div className={styles.text}>
      <p>{props.message}</p>
    </div>
  </div>
);

ErrorComponent.propTypes = {
  message: PropTypes.string.isRequired
};

export default ErrorComponent;
