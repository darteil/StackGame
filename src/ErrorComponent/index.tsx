import React from 'react';
import styles from './styles.css';

interface IProps {
  message: string;
}

const ErrorComponent = (props: IProps) => (
  <div className={styles.wrap}>
    <div className={styles.text}>
      <p>{props.message}</p>
    </div>
  </div>
);

export default ErrorComponent;
