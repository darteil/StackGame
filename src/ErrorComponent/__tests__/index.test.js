import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ErrorComponent from 'App/ErrorComponent';


describe('ErrorComponent', () => {
  it('should render correctly', () => {
    const output = shallow(
      <ErrorComponent message="error message" />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
