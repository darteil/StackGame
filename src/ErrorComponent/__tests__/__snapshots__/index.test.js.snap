// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent should render correctly 1`] = `
<div>
  <div>
    <p>
      error message
    </p>
  </div>
</div>
`;
